optaplanner-core/src/main/java/org/optaplanner/core/config/solver/EnvironmentMode.java

```diff
@@ -47,7 +47,7 @@
     FULL_ASSERT,
     /**
      * This mode turns on several assertions (but not all of them)
-     * to fail-fast on a bug in a {@link Move} implementation, a score rule, the rule engine itself or something else
+     * to fail-fast on a bug in a {@link Move} implementation, a constraint, the engine itself or something else
      * at a horrible performance cost.
```

Comment:
```suggestion
     * to fail-fast on a bug in a {@link Move} implementation, a constraint, the engine itself or something else
     * at an overwhelming performance cost.
```

Somehow appears more professional to me.